servlet-engine/src/main/java/io/micronaut/servlet/engine/body/ByteQueue.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.servlet.engine.body;
+
+import io.micronaut.core.annotation.Internal;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Queue;
+
+/**
+ * Non-thread-safe queue for bytes.
+ *
+ * @author Jonas Konrad
+ * @since 4.9.0
+ */
+@Internal
+final class ByteQueue {
```

Comment:
can be a record

Reply:
I want to hide the queue field.

Also there is a pretty obvious more efficient implementation (based on a byte array and counters) that I want to keep the option open for.